### PR TITLE
feat(proposal): add validation to start time

### PR DESCRIPTION
**Pull request by Todmy**
### Issues
[#334
](https://github.com/snapshot-labs/snapshot-hub/issues/334)

Fixes #
Set starting date not greater than the current time. Restrict users to enter lower time than the current time

### How to test
1. Create Proposal
2. Try to enter the start date/time lower than the current time

### To-Do
- [ ] 

### Self-review checklist
- [x] I have performed a full self-review of my changes
- [ ] I have tested my changes on a preview deployment
- [x] I have tested my changes on different screen sizes (sm, md)
- [x] I have tested my changes on a custom domain

### Additional notes or considerations
*_(Include any other relevant information or context that may be helpful for the reviewer)_*

### CODE REVIEW

File: src/views/SpaceCreate.vue [hunk 1]
```diff
@@ -159,7 +159,11 @@ function getFormattedForm() {
     .map(choice => choice.text)
     .filter(choiceText => choiceText.length > 0);
   updateTime();
-  clonedForm.start = dateStart.value;
+  const thisMomentTimestamp = parseInt((Date.now() / 1e3).toFixed());
+  clonedForm.start =
+    dateStart.value < thisMomentTimestamp
```
Comment (wa0x6e) [inline src/views/SpaceCreate.vue:164]: isn't `Math.max(thisMomentTimestamp, dateStart.value)` better ?

Comment (0508511812) [reply src/views/SpaceCreate.vue]: image

Comment (0508511812) [reply src/views/SpaceCreate.vue]: image

**Comment by samuveth** [comment]
How does this fix the [issue](https://github.com/snapshot-labs/snapshot-hub/issues/334)? It says "enforce start date higher or equal to creation date for proposal", but I only see that you are preventing start date in the past

File: src/components/ModalSelectDate.vue [hunk 2]
```diff
@@ -39,6 +40,17 @@ watch(open, () => {
   time.value = `${h}:${m}`;
   input.value = dateString;
 });
+
+watchEffect(() => {
```
Comment (samuveth) [inline src/components/ModalSelectDate.vue:44]: Why is this needed now?

Comment (samuveth) [reply src/components/ModalSelectDate.vue:44]: Okay I see now, but when I enter a time in the past the input is no longer functioning

Comment (Todmy) [reply src/components/ModalSelectDate.vue:44]: could you specify the way you did it? This code should prevent entering past time.
p.s.: for more clever users I added an extra check [here](https://github.com/snapshot-labs/snapshot/pull/3991/files#diff-2ac25a10c6190e76b7a330fbd6a8438aa8af90245144b2a2e3abba7d5fed8224R162) and [here](https://github.com/snapshot-labs/snapshot-sequencer/pull/71/files#diff-4cd70d1c07b79fdcdf325dd8afe5e8e6af488c09ff1b19a1bf54c1825e349576R105) :)

Comment (samuveth) [reply src/components/ModalSelectDate.vue:44]: https://www.loom.com/share/e95637017f6c4c9a932fac2fe7a5dda0

Comment (samuveth) [reply src/components/ModalSelectDate.vue:44]: Instead of filling the input automatically with the current time, we should let the user enter time in that past and clicks "Select" we should show a warning and prevent "Select"

**Review by samuveth — APPROVED** [review]
tACK
